lib/ocpp/v201/smart_charging.cpp

```diff
@@ -423,380 +423,73 @@ int get_requested_limit(const int limit, const int nr_phases, const ChargingRate
-                    temp_period_end_time = period_date_time_pair.end_time;
-
-                    temp_number_phases = period.numberPhases.value_or(DEFAULT_AND_MAX_NUMBER_PHASES);
-
-                    ChargingSchedule first_charging_schedule = profile.chargingSchedule.front();
-
-                    float limit =
-                        get_power_limit(period.limit, temp_number_phases, first_charging_schedule.chargingRateUnit);
-                    int32_t stackLevel = profile.stackLevel;
-
-                    EVLOG_info << "period.has_value() limit = " << limit;
-                    EVLOG_info << "period.has_value() stackLevel = " << limit;
-
-                    // update data structure with limit and stack level for this profile
-                    current_purpose_and_stack_limits.at(profile.chargingProfilePurpose).limit = limit;
-                    current_purpose_and_stack_limits.at(profile.chargingProfilePurpose).stack_level = stackLevel;
-                }
-            }
-        };
-
-        // if there is a limit with purpose TxProfile it overrules the limit of purpose TxDefaultProfile
-        if (current_purpose_and_stack_limits.at(ChargingProfilePurposeEnum::TxProfile).limit != MAX_PERIOD_LIMIT) {
-            significant_limit_stack_level_pair =
-                current_purpose_and_stack_limits.at(ChargingProfilePurposeEnum::TxProfile);
-        } else {
-            significant_limit_stack_level_pair =
-                current_purpose_and_stack_limits.at(ChargingProfilePurposeEnum::TxDefaultProfile);
+    std::optional<ocpp::DateTime> session_start{};
+
+    // 1.6 Connector->Transaction->StampedEnergyWh
+    // if (const auto& itt = connectors.find(connector_id); itt != connectors.end()) {
+    //     // connector exists!
+    //     if (itt->second->transaction) {
+    //         session_start.emplace(ocpp::DateTime(
+    //             floor<seconds>(itt->second->transaction->get_start_energy_wh()->timestamp.to_time_point())));
+    //     }
+    // }
+
+    // 2.0 EvseInterface->get_transaction()->EnhancedTransaction->get_transaction()->Transaction
+    // NOTE: No longer works because now using EVSEManager
+    // if (const auto& itt = this->evses.find(evse_id); itt != this->evses.end()) {
+    //     if (itt->second->get_transaction()) {
+    //         // const ocpp::DateTime now(date::utc_clock::now());
+    //         // EVLOG_info << "   2nd> has transation" << itt->second->get_transaction()->get_transaction();
+    //         session_start.emplace(start_time);
+    //     }
+    // }
+
+    std::vector<period_entry_t> charge_point_max{};
+    std::vector<period_entry_t> tx_default{};
+    std::vector<period_entry_t> tx{};
+
+    for (const auto& profile : valid_profiles) {
+        std::vector<period_entry_t> periods{};
+        periods = ocpp::v201::calculate_profile(start_time, end_time, session_start, profile);
+
+        for (auto period : periods) {
+            EVLOG_info << "period> " << period;
         }
```

Comment:
Is this for loop just to log the periods? Would:


```suggestion
        EVLOG_info << "periods> " << periods;
```

work?

Reply:
No, but I'll remove it